payments/db/sql_store.go

```diff
@@ -143,23 +147,109 @@ var _ DB = (*SQLStore)(nil)
 // including attempts, hops, and custom records from the database.
 // This is a convenience wrapper around the batch loading functions for single
 // payment operations.
-func (s *SQLStore) fetchPaymentWithCompleteData(ctx context.Context,
-	db SQLQueries, dbPayment sqlc.PaymentAndIntent) (*MPPayment, error) {
+func fetchPaymentWithCompleteData(ctx context.Context,
+	cfg *sqldb.QueryConfig, db SQLQueries,
+	dbPayment sqlc.PaymentAndIntent) (*MPPayment, error) {
 
 	payment := dbPayment.GetPayment()
 
 	// Load batch data for this single payment.
-	batchData, err := s.loadPaymentsBatchData(ctx, db, []int64{payment.ID})
+	batchData, err := batchLoadPaymentsRelatedData(
+		ctx, cfg, db, []int64{payment.ID},
+	)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load batch data: %w", err)
 	}
 
 	// Build the payment from the batch data.
-	return s.buildPaymentFromBatchData(dbPayment, batchData)
+	return buildPaymentFromBatchData(dbPayment, batchData)
+}
+
+// paymentsAggregate holds the full payment data when batch loading core
+// payment data and all the related data for a payment.
+type paymentsAggregate struct {
```

Comment:
nit: `fullPaymentData` or `paymentCompleteData`

Reply:
went with `paymentsCompleteData` with payments instead of payment because it includes multiple payments